csrc/sys_utils.cpp

```diff
@@ -162,6 +166,28 @@ LibraryLoader::~LibraryLoader() {
   }
 }
 
+namespace {
+
+// Callback should return 0 to continue iterationg. A non-zero return
+// value would stop the iteration.
+int detectComputeSanitizerCallback(
+    struct dl_phdr_info* info,
+    size_t size,
+    void* data) {
+  std::string lib_name = info->dlpi_name;
+  if (lib_name.find("compute-sanitizer") != std::string::npos) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+} // namespace
+
+bool detectComputeSanitizer() {
+  return dl_iterate_phdr(detectComputeSanitizerCallback, nullptr) != 0;
+}
+
 } // namespace nvfuser
 
 #else
```

Comment:
```suggestion
#else // #if defined(__linux__)
```

Reply:
It took me many clicks to find the corresponding if.

Reply:
Hahaha, yeah, that actually did hit me first when testing this new function. Thanks for the suggestion!